### ISSUE TITLE
Connection is connected again and again

### ISSUE THREAD

**Issue by PengsreangSrun** [socketio/socket.io-client-swift#101]
Can you solve me this problem because i have met following problem: This is my log:
2015-06-19 10:33:49.286  SocketClient: Adding handler for event: connect
2015-06-19 10:33:49.286  SocketClient: Adding engine
2015-06-19 10:33:49.286 SocketEngine: Starting engine
2015-06-19 10:33:49.287  SocketEngine: Handshaking
2015-06-19 10:33:49.943 SocketEngine: Got message: 0{"sid":"gGIGh932oH7Fqx_DAAAo","upgrades":[],"pingInterval":500,"pingTimeout":60000}
2015-06-19 10:33:49.944  SocketEngine: Got message: 40
2015-06-19 10:33:49.945  SocketParser: Parsing 0
2015-06-19 10:33:49.945  SocketParser: Decoded packet as: SocketPacket {type: 0; data: nil; id: nil; placeholders: nil;}
2015-06-19 10:33:49.945 SuperChat[5268:590285] SocketEngine: Writing ws: :nil
2015-06-19 10:33:49.945 SuperChat[5268:590285] SocketEngine: Sending ws:  as type: 2
2015-06-19 10:33:49.946 SuperChat[5268:590285] SocketClient: Handling event: reconnect with data: (
    "Ping timeout"
)
2015-06-19 10:33:49.946  SocketClient: Starting reconnect
2015-06-19 10:33:49.946  SocketClient: Trying to reconnect
2015-06-19 10:33:49.946  SocketClient: Handling event: reconnectAttempt with data: (
    5
)
2015-06-19 10:33:49.946  SocketClient: Adding engine
2015-06-19 10:33:49.946  SocketClient: Socket connected
2015-06-19 10:33:49.947  SocketEngine: Starting engine
2015-06-19 10:33:49.947  SocketEngine: Handshaking
2015-06-19 10:33:49.947  SocketEngine: Engine is being deinit
2015-06-19 10:33:49.947  SocketClient: Handling event: connect with data: 
2015-06-19 10:33:50.285  TestFairy: Session started successfully
2015-06-19 10:33:50.474  SocketEngine: Got message: 0{"sid":"yqK1-aOfAdS2JEfrAAAp","upgrades":[],"pingInterval":500,"pingTimeout":60000}
2015-06-19 10:33:50.475  SocketEngine: Got message: 40
2015-06-19 10:33:50.475  SocketParser: Parsing 0
2015-06-19 10:33:50.476 SuperChat[5268:590305] SocketParser: Decoded packet as: SocketPacket {type: 0; data: nil; id: nil; placeholders: nil;}
2015-06-19 10:33:50.477  SocketClient: Socket connected
2015-06-19 10:33:50.477  SocketClient: Handling event: connect with data: 
2015-06-19 10:33:50.478  SocketEngine: Writing ws: :nil
2015-06-19 10:33:50.478  SocketEngine: Sending ws:  as type: 2
2015-06-19 10:33:50.479  SocketClient: Handling event: reconnect with data: (
    "Ping timeout"
)
2015-06-19 10:33:50.479  SocketClient: Starting reconnect
2015-06-19 10:33:50.480  SocketClient: Trying to reconnect
2015-06-19 10:33:50.480  SocketClient: Handling event: reconnectAttempt with data: (
    5
)
2015-06-19 10:33:50.481  SocketClient: Adding engine
2015-06-19 10:33:50.481  SocketEngine: Starting engine
2015-06-19 10:33:50.482  SocketEngine: Handshaking
2015-06-19 10:33:50.482  SocketEngine: Engine is being deinit
2015-06-19 10:33:50.993  SocketEngine: Got message: 0{"sid":"4C7i5c45t--jtahnAAB4","upgrades":[],"pingInterval":500,"pingTimeout":60000}
2015-06-19 10:33:50.994  SocketEngine: Writing ws: :nil
2015-06-19 10:33:50.994  SocketEngine: Sending ws:  as type: 2
2015-06-19 10:33:50.994  SocketEngine: Got message: 40
2015-06-19 10:33:50.994  SocketParser: Parsing 0
2015-06-19 10:33:50.994  SocketParser: Decoded packet as: SocketPacket {type: 0; data: nil; id: nil; placeholders: nil;}
2015-06-19 10:33:50.994  SocketClient: Socket connected
2015-06-19 10:33:50.995  SocketClient: Handling event: connect with data:

**Comment by nuclearace** [comment]
Have you tried creating the socket with forcePolling

**Comment by PengsreangSrun** [comment]
I tried both of forcePolling and forceWebsocket, but it was useless!

**Comment by nuclearace** [comment]
Would it be possible to create a small project that reproduces it? Because I'm having trouble reproducing it.

**Comment by nuclearace** [comment]
Any more on this issue?

**Comment by alexpersian** [comment]
I am observing the same issues in a current project using this Socket.IO Swift library.
Multiple connected, disconnect, reconnects without any code or user prompts.

**Comment by alexpersian** [comment]
Hmm, was there any resolution to this issue?

**Comment by nuclearace** [comment]
I haven't been able to reproduce it.

**Comment by mephi1984** [comment]
I got same issue.

Brief research shows me, that issue is in SocketEngine.swift file. 
```swift
    private func sendPing() {
        guard connected else { return }

        // Server is not responding
        if pongsMissed > pongsMissedMax {
            client?.engineDidClose(reason: "Ping timeout")
            
            return
        }
```

In my case, _pongsMissed_ always 4, _pongsMissedMax_ is 3, that's why connection drops. However, on reconnect, _pongsMissed_ never reset to 0, so connection drops, then reconnect again, then drops again, then reconnect again, and so on.

I added workaround, by resetting _pongsMissed_ to 0 at same place when _connected_ is set to true, totally 2 search matches.

Do I need to submit pull request?
